nativelib/src/main/scala/scala/scalanative/runtime/Backtrace.scala

```diff
@@ -0,0 +1,236 @@
+package scala.scalanative.runtime
+
+import java.io.File
+import scala.scalanative.runtime.dwarf.BinaryFile
+import java.io.RandomAccessFile
+import scala.scalanative.runtime.dwarf.MachO
+import scala.scalanative.runtime.dwarf.DWARF
+import scala.collection.mutable
+import scala.scalanative.unsafe.CSize
+import scala.scalanative.unsafe.Tag
+import scala.scalanative.unsigned.UInt
+import scala.collection.concurrent.TrieMap
+import scala.scalanative.unsafe.Zone
+import scalanative.unsigned._
+import scala.io.Source
+import scala.scalanative.runtime.dwarf.DWARF.CompileUnit
+import scala.annotation.tailrec
+import scala.scalanative.runtime.dwarf.DWARF.DIE
+import scala.util.Using
+import scala.util.Failure
+import scala.util.Success
+
+object Backtrace {
+  private sealed trait Format
+  private case object MACHO extends Format
+  private case object ELF extends Format
+  private case class DwarfInfo(
+      subprograms: IndexedSeq[SubprogramDIE],
+      strings: DWARF.Strings,
+      /** ASLR offset (minus __PAGEZERO size for macho) */
+      offset: Long,
+      format: Format
+  )
+
+  private case class SubprogramDIE(
+      lowPC: Long,
+      highPC: Long,
+      line: Int,
+      filenameAt: Option[UInt]
+  )
+
+  private val MACHO_MAGIC = "cffaedfe"
+  private val ELF_MAGIC = "7f454c46"
+
+  private val cache = TrieMap.empty[String, Option[DwarfInfo]]
+
+  def decodeFileline(pc: Long): Option[(String, Int)] = {
+    cache.get(filename) match {
+      case Some(None) =>
+        None // cached, there's no debug section
```

Comment:
cache the debug information, so we don't need to try to parse the executable more than once.